kernel/exec.c

```diff
@@ -616,6 +616,17 @@ int __do_execve(const char *file, struct exec_args argv, struct exec_args envp)
 
     current->did_exec = true;
     vfork_notify(current);
+
+    if (current->traced) {
+        lock(&pids_lock);
```

Comment:
unnecessary?

Reply:
Taking the lock, or sending a signal? I think both are necessary…

Reply:
Taking the lock, but no need to discuss it here, there are like 500 other comments about it